test-harness/init.sh

```diff
@@ -97,5 +102,10 @@ function template_build_targets() {
         esac
     done
 
+    if [ ${#TEST_RUN_MAP[@]} -eq 0 ]; then
+        echo "INFO: No templates to process. Exiting build step"
+        exit 0
+    fi
```

Comment:
Is this necessary? Without it, will the tests harness just loop over an empty array and produce no results anyways?

Reply:
if there's no template directories to include in the run then we should exit and skip rebuilding the docker image.